src/runtime/api-channel.ts

```diff
@@ -32,15 +32,25 @@ interface MappingInfo {
   redundant?: boolean;
   value?: MappingInfo;
   key?: MappingInfo;
-  converter?: Literalizer;
+  converter?: Literalizer | LiteralizerParticleSpec | LiteralizerPropagatedException;
   identifier?: boolean;
   ignore?: boolean;
 }
 
 // TODO(shans): are there better types that I can use for this?
 interface Literalizer {
-  prototype: {toLiteral: () => {}};
-  fromLiteral: ({}) => {};
+  prototype: {toLiteral: () => TypeLiteral};
+  fromLiteral: (typeliteral: TypeLiteral) => Type;
+}
+
```

Comment:
This is a little hacky, but describes the current usage.  Should ParticleSpec/PropagatedException implement Type/TypeLiteral instead?

Reply:
I tried to parameterize Literalizer, but none of them survived the use of decorators with any fidelity.

I also started down the path of creating a new Literalizable interface but ran into the same issues with adding Types to EnityType (namely lack of static methods in interfaces https://github.com/Microsoft/TypeScript/issues/14600)  Especially with generics.

Since the latter solution is preferred, but also much more invasive I'll spin those changes out to another cleanup issue.